flink-runtime/src/main/scala/org/apache/flink/runtime/jobmanager/JobManager.scala

```diff
@@ -1024,6 +1024,17 @@ class JobManager(
 
     case RequestWebMonitorPort =>
       sender() ! ResponseWebMonitorPort(webMonitorPort)
+
+    case MetricRequest =>
+      metricsRegistry match {
+        case Some(registry) =>
+          registry.getQueryService match {
+            case Some(queryService) =>
+              queryService ! sender()
+            case None =>
+          }
+        case None =>
```

Comment:
In both `None` cases you should return a failure message to the sender so that it does not have to wait until the future times out.